sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -185,85 +184,45 @@ public void updateAddresses(final URI serverKey) {
-            // Due to the fact that forcing the CollectionRoutingMap to be refreshed in Gateway there is additional
-            // load on the ServiceFabric naming service we want to throttle how often we would force the collection
-            // routing map refresh
-            // These are the throttle conditions: We will only enforce collection routing map to be refreshed
-            // - if there has been at least 1 attempt to just refresh replica addresses without forcing collection
-            //   routing map refresh on the physical partition before
-            // - only one request per Container to force collection routing map refresh is allowed every 30 seconds
-            //
-            // The throttling logic is implemented in  `ForcedRefreshMetadata.shouldIncludeCollectionRoutingMapRefresh`
-            ForcedRefreshMetadata forcedRefreshMetadata = this.lastForcedRefreshMap.computeIfAbsent(
-                partitionKeyRangeIdentity.getCollectionRid(),
-                (colRid) -> new ForcedRefreshMetadata());
-
-            if (request.forceCollectionRoutingMapRefresh) {
-                forcedRefreshMetadata.signalCollectionRoutingMapRefresh(
-                    partitionKeyRangeIdentity,
-                    true);
-            } else if (forcedRefreshMetadata.shouldIncludeCollectionRoutingMapRefresh(partitionKeyRangeIdentity)) {
-                request.forceCollectionRoutingMapRefresh = true;
-                forcedRefreshMetadata.signalCollectionRoutingMapRefresh(
-                    partitionKeyRangeIdentity,
-                    true);
-            } else {
-                forcedRefreshMetadata.signalPartitionAddressOnlyRefresh(partitionKeyRangeIdentity);
-            }
-        } else if (request.forceCollectionRoutingMapRefresh) {
-            ForcedRefreshMetadata forcedRefreshMetadata = this.lastForcedRefreshMap.computeIfAbsent(
-                partitionKeyRangeIdentity.getCollectionRid(),
-                (colRid) -> new ForcedRefreshMetadata());
-            forcedRefreshMetadata.signalCollectionRoutingMapRefresh(
-                partitionKeyRangeIdentity,
-                false);
-        }
-
         logger.debug("PartitionKeyRangeIdentity {}, forceRefreshPartitionAddresses {}",
             partitionKeyRangeIdentity,
             forceRefreshPartitionAddresses);
         if (StringUtils.equals(partitionKeyRangeIdentity.getPartitionKeyRangeId(),
-                PartitionKeyRange.MASTER_PARTITION_KEY_RANGE_ID)) {
+            PartitionKeyRange.MASTER_PARTITION_KEY_RANGE_ID)) {
 
             // if that's master partition return master partition address!
             return this.resolveMasterAsync(request, forceRefreshPartitionAddresses, request.properties)
                        .map(partitionKeyRangeIdentityPair -> new Utils.ValueHolder<>(partitionKeyRangeIdentityPair.getRight()));
         }
 
+        evaluateCollectionRoutingMapRefreshForServerPartition(
```

Comment:
This call will still fail for Master partition, since the collectionRid will be null here. 
Also, if we are throwing error or collectionRid being null, then we also need to fix the GatewayAddressCacheTest.java

Reply:
the if-statement for MasterPartition will return above - so this shouldn't be an issue.

Reply:
Missed the if part, thanks for clarifying it.